src/gui/proxyauthhandler.cpp

```diff
@@ -190,9 +189,7 @@ bool ProxyAuthHandler::getCredsFromKeychain()
         _readPasswordJob->setInsecureFallback(false);
         _readPasswordJob->setKey(keychainPasswordKey());
         _readPasswordJob->setAutoDelete(false);
-        connect(_readPasswordJob.data(), &QKeychain::Job::finished,
-            this, &ProxyAuthHandler::slotKeychainJobDone);
-        _keychainJobRunning = true;
+        connect(_readPasswordJob.data(), &QKeychain::Job::finished, &waitLoop, &QEventLoop::quit);
```

Comment:
This is done twice, this is better done in a function "execJob()" or something like that. This would avoid the code duplication.

Bonus point if it is template based to make it work with QDialogs as well since the first QEventLoop in that file is about a QDialog. :-)